crates/nassun/src/resolver.rs

```diff
@@ -230,6 +231,20 @@ impl PackageResolver {
                 versions: packument.versions.keys().map(|k| k.to_string()).collect(),
             })
             .and_then(|v| {
+                if let Some(deprecated) = &v.deprecated {
+                    tracing::warn!(
+                        "{} {}@{}: {}",
+                        "deprecated".magenta(),
+                        name,
+                        v.manifest
+                            .version
+                            .as_ref()
+                            .map(|v| v.to_string())
+                            .unwrap_or_else(|| "unknown".into()),
+                        deprecated
+                    );
+                }
```

Comment:
I think this is the wrong place for this warning. You should move this to `node-maintainer` itself, and I think you'll definitely be able to `ProgressBar::suspend` without much fuss, if you do it that way? The message itself otherwise looks fine.